include/deal.II/meshworker/scratch_data.h

```diff
@@ -624,6 +624,107 @@ namespace MeshWorker
 
     /** @} */ // NeighborCellMethods
 
+    /**
+     * @name hp-compatible methods to work on cells and neighbor cells
+     */
+    /** @{ */ // hpCellMethods
+
+    /**
+     * Initialize the internal FEFaceValues to use the given @p face_no on the given
+     * @p cell, and return a reference to it.
+     *
+     * This variant of the reinit() function compares the active finite element
+     * on each cell, and chooses the dominated finite element's index to select
+     * the quadrature rule and mapping for the returned FEFaceValues object.
+     * This is useful in instances where the shape functions of the elements on
+     * either side of a face are being evaluated at the same time (such as is
+     * done in DG methods). See FECollection::find_dominated_fe() for more
+     * information on the selection process.
+     *
+     * After calling this function, get_current_fe_values() will return the
+     * same object of this method, as an FEValuesBase reference.
+     */
+    const FEFaceValues<dim, spacedim> &
+    reinit(const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell,
+           const typename DoFHandler<dim, spacedim>::active_cell_iterator
+             &                neighbor_cell,
+           const unsigned int face_no);
+
+    /**
+     * Initialize the internal FESubfaceValues to use the given @p subface_no,
+     * on @p face_no, on the given @p cell, and return a reference to it.
+     *
+     * This variant of the reinit() function compares the active finite element
+     * on each cell, and chooses the dominated finite element's index to select
+     * the quadrature rule and mapping for the returned FEFaceValues object.
+     * This is useful in instances where the shape functions of the elements on
+     * either side of a face are being evaluated at the same time (such as is
+     * done in DG methods). See FECollection::find_dominated_fe() for more
+     * information on the selection process.
+     *
+     * After calling this function, get_current_fe_values() will return the
+     * same object of this method, as an FEValuesBase reference.
+     *
+     * If @p subface_no is numbers::invalid_unsigned_int, the reinit() function
+     * that takes only the @p cell and the @p face_no is called.
+     */
```

Comment:
I was wondering why I'd do this, and I now see why: this mirrors the existing functions that `ScratchData` already has. See https://dealii.org/developer/doxygen/deal.II/classMeshWorker_1_1ScratchData.html#a8cc613e7f254fdbd076429a0c760d0e3 and https://dealii.org/developer/doxygen/deal.II/classMeshWorker_1_1ScratchData.html#ae7972a025f7166ca5ae3b6a24f65c54c . I'm happy to do as you suggest, but it would result in an inconsistent interface. How strongly do you feel about this? We could also do the same for the other functions, and maybe put all of the refactoring in a follow-up PR?

Reply:
That's fine to keep it as is then.

Reply:
Ok, thanks a lot! I've addressed all of your other comments.